junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/TestWatcher.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api.extension;
+
+import java.util.Optional;
+
+public interface TestWatcher extends Extension {
+
+	default void testSuccessful(String identifier, ExtensionContext context) {
+	}
+
+	default void testAborted(String identifier, Throwable cause, ExtensionContext context) {
+	}
+
+	default void testFailed(String identifier, Throwable cause, ExtensionContext context) {
+	}
+
+	default void testSkipped(String identifier, Optional<String> Reason, ExtensionContext context) {
```

Comment:
After putting a bit more thought into this, I am inclined to say that we should name this `testDisabled()` instead of `testSkipped()`.

Although "skipped" is technically correct with regard to the `Node` abstraction at the engine level, the actual terminology used within JUnit Jupiter is "disabled" due to the `org.junit.jupiter.api.extension.ConditionEvaluationResult.isDisabled()` API.

Thus, I think it is best to be consistent regarding the terminology used within the Jupiter extension model.

@junit-team/junit-lambda, thoughts?

Reply:
Based on the 👍 from @marcphilipp, please do rename that to `testDisabled()`.